ls1intum/Artemis
# Prevent other tutors from assessing submissions of a tutor's teams

### Checklist

### Motivation and Context
This PR addresses [this](https://github.com/ls1intum/Artemis/issues/1543) issue.

### Description
- Hide "next submission" button for team exercise assessment
- Consider all submissions (also non-submitted ones) for assessment
- Implement authorization for submitting assessments for team exercises (only the team tutor or instructors can assess the submission of a team)

### Steps for Testing
1. Log in to Artemis as a Tutor
2. Navigate to Course Management
3. Click on a Course with a Team Exercise
4. Click on Tutor Course Dashboard
5. Click on Exercise Teams for an exercise
6. Click on the short name of one of your teams
7. Assess one of the submissions
8. Make sure that there is no green button "Assess Next Submission"
9. Check that you can save and submit an assessment

**Note:** If you experience server errors in relation to complaints or more feedback requests for teams, this can happen if you had previously deployed [this](https://github.com/ls1intum/Artemis/pull/1527) PR (which is not merged into develop yet) and persisted data that is not yet handled in develop. So in that case an error is to be expected.

### Screenshots
![OnPaste 20200602-000714](https://user-images.githubusercontent.com/7109225/83459615-06be2c00-a465-11ea-8f69-b25e230a64c9.png)
